datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -1634,6 +1638,99 @@ pub(crate) fn asymmetric_join_output_partitioning(
         }
     }
 }
+#[derive(Debug, Clone)]
+/// store physicalexpr about the dynamic filters
+pub struct PhysicalDynamicFiltersInfo {
+    pub probe_side_columns: Vec<Arc<dyn PhysicalExpr>>,
+    pub aggregates: Vec<Arc<dyn PhysicalExpr>>,
+    // final predicates we should execute
+    final_expr: Option<Arc<dyn PhysicalExpr>>,
+}
+
+impl PhysicalDynamicFiltersInfo {
+    pub fn new(
+        probe_side_columns: Vec<Arc<dyn PhysicalExpr>>,
+        aggregates: Vec<Arc<dyn PhysicalExpr>>,
+    ) -> Self {
+        Self {
+            probe_side_columns,
+            aggregates,
+            final_expr: None,
+        }
+    }
+
+    pub fn with_final_expr(mut self, expr: Arc<dyn PhysicalExpr>) -> Self {
+        self.final_expr = Some(expr);
+        self
+    }
+
+    pub fn filter_batch(
+        &mut self,
+        records: &RecordBatch,
+    ) -> Result<RecordBatch, DataFusionError> {
+        let filter_expr = if let Some(final_expr) = self.final_expr.as_ref() {
+            // If final_expr already exists, use it directly
+            final_expr.clone()
+        } else {
+            // Otherwise, compute the new filter expression
+            let mut new_filter_expr: Option<Arc<dyn PhysicalExpr>> = None;
+
+            for (i, col) in self.probe_side_columns.iter().enumerate() {
+                let min_value = self.aggregates[i].evaluate(records)?;
```

Comment:
Hm... I think you want this be evaluated on other side of the join once (after loading the build side) and then be passed on to the other side.

Now the filter seems to be created from the probe side and the filter is evaluated from probe side?